classlib/Database.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+/**
+ * Database class - create single instance
+ *
+ * @package Database
+ * @subpackage
+ * @category
+ * @author Sanjeeb Rao
+ */
+
+class Database {
+    private $db = array(
+        'master' => array(
+            'host' => 'localhost',
+            'username' => 'root',
+            'password' => 'mindfire',
+            'db_name' => 'registration'
+        ),
+        'slave' => array(
+            'host' => 'localhost',
+            'username' => 'root',
+            'password' => 'mindfire',
+            'db_name' => 'registration'
+        )
+    );
+
+    private $connection = NULL;
+
+    // The single instance
+    private static $instance = NULL;
+
+    /**
+     * Creating connection.
+     *
+     * @access private constructor of Database
+     * @param  void
+     */
+    private function __construct()
+    {
+        $this->connection = new mysqli(
+            $this->db['master']['host'],
+            $this->db['master']['username'],
+            $this->db['master']['password'],
+            $this->db['master']['db_name']);
+
+        // Error handling
+        if(mysqli_connect_error())
+        {
+            header('Location: error.php');
+        }
+    }
+
+    /**
+     * Creating connection.
+     *
+     * @access private
+     * @param  void
+     */
+    private function __clone()
+    {
+    }
+
+    /**
+     * Get an instance of the Database.
+     *
+     * @access public get_instance
+     * @param  void
+     * @return Instance
+     */
+    public static function get_instance()
+    {
+        if(self::$instance === NULL)
+        {
+
+            // If no instance then make one
+            self::$instance = new self();
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * Retrun connection.
+     *
+     * @access public  get_connection
```

Comment:
Should be only @access public. This change can also be done at similar other places.